libs/domain/picking/src/components/picking-header/picking-header.component.ts

```diff
@@ -56,12 +69,23 @@ export class PickingHeaderComponent extends PickingListMixin(LitElement) {
           contentBehavior: 'modal',
           label: i18n('oryx.picking.account'),
         }}
-      ></oryx-site-navigation-item>`;
+      ></oryx-site-navigation-item>
+      <oryx-discard-picking
+        @oryx.close=${this.closeDiscardModal}
+        @oryx.back=${this.back}
+      ></oryx-discard-picking>`;
+  }
+
+  protected openDiscardModal(): void {
+    this.discardModal?.toggleAttribute('open', true);
+  }
+
+  protected closeDiscardModal(): void {
+    this.discardModal?.toggleAttribute('open', false);
   }
 
   protected back(): void {
-    //TODO - display discard modal
-    this.routerService.back();
+    this.routerService.back(this.pickingList?.cartNote ? 2 : 1);
```

Comment:
we did discuss that we should not go back 2 or 1 times, as it's unreliable, and maybe not needed, anything changed?